src/components/splashScreen/TitleBadge.tsx

```diff
@@ -7,17 +7,31 @@ type TitleBadgeProps = {
 };
 
 export default function TitleBadge(props: TitleBadgeProps) {
-  const { isPaniniOrdered } = useOutletContext() as LayoutContext;
+  const { userStep, setOrderData } = useOutletContext() as LayoutContext;
   const navigate = useNavigate();
   const handleClick = () => {
-    navigate(`${isPaniniOrdered ? "/" : "/panini_creator"}`);
+    let destination = "/";
+    switch (userStep) {
+      case 0:
+        destination = "/form_transition/without_order";
```

Comment:
**nitpick:** This would be extremely difficult to test, and maintain. You should be able to pass `onClick` to this component instead of hardcoding this routing logic.